src/Engine/ProtoCore/RuntimeCore.cs

```diff
@@ -378,4 +378,54 @@ public void RemoveExecutedAstGuids()
             executedAstGuids.Clear();
         }
     }
+
+    internal class MSILRuntimeCore
+    {
+        // The global class table and function tables
+        private ClassTable ClassTable { get; set; }
+        private ProcedureTable ProcTable { get; set; }
+        private ProcedureNode ProcNode { get; set; }
+        private TypeSystem TypeSystem { get; set; }
+        private InternalAttributes internalAttributes { get; set; }
+
+        protected MSILRuntimeCore()
+        {
+            ClassTable = new ClassTable();
+            TypeSystem = new TypeSystem();
+            TypeSystem.SetClassTable(ClassTable);
+            ProcNode = null;
+            ProcTable = new ProcedureTable(Constants.kGlobalScope);
+
+            // Initialize internal attributes
+            internalAttributes = new InternalAttributes(ClassTable);
+        }
+
+        private static MSILRuntimeCore instance;
+        internal static MSILRuntimeCore Instance => instance == null ? new MSILRuntimeCore() : instance;
+
+        internal bool ConvertibleTo(CLRStackValue from, Type to)
+        {
+            return ClassTable.ClassNodes[from.TypeUID]?.ConvertibleTo(to.UID) ?? false;
+        }
+
+        internal void LogWarning(Runtime.WarningID ID, string message)
+        {
+            System.Console.WriteLine($"{ID}{message}");
+        }
+
+        internal int GetProtoCoreTypeID(System.Type type)
+        {
+            int typeUID = ClassTable.IndexOf(CLRObjectMarshaler.GetTypeName(type));
+            if (typeUID == ProtoCore.DSASM.Constants.kInvalidIndex)
+            {
+                typeUID = CLRModuleType.GetProtoCoreType(type, null).UID;
+            }
+            return typeUID;
+        }
+
+        internal int GetCoercionScore(int srcType, int targetType)
+        {
+            return ClassTable?.ClassNodes[srcType]?.GetCoercionScore(targetType) ?? (int)ProcedureDistance.NotMatchScore;
```

Comment:
You said you're not populating the classtable currently for `MSILRuntimeCore` and are using the old VM's classtable, correct? Can you point to the place in this PR where you're doing so?

Reply:
Look at the method above this one
`GetProtoCoreTypeID`
```
            int typeUID = ClassTable.IndexOf(CLRObjectMarshaler.GetTypeName(type));
            if (typeUID == ProtoCore.DSASM.Constants.kInvalidIndex)
            {
                typeUID = CLRModuleType.GetProtoCoreType(type, null).UID;
            }
```
`CLRModuleType.GetProtoCoreType(type, null)`  - CLRModuleType has most of the types loaded by the old VM

![image](https://user-images.githubusercontent.com/46732933/182870808-c2a77e14-1712-4a54-8498-957b50bdef7b.png)